playbooks/roles/backups/templates/backup.py

```diff
@@ -0,0 +1,324 @@
+    if os.path.isfile(backup_tar):
+        os.remove(backup_tar)
+
+    if os.path.isdir(backup_path):
+        shutil.rmtree(backup_path)
+    elif os.path.isfile(backup_path):
+        os.remove(backup_path)
+
+
+def restore(service_name, backup_path, uncompress=True):
+    """
+    Restore a database from a backup.
+
+        service_name: The name of the service whose database is to be restored,
+            either mysql or mongodb.
+        backup_path: The absolute path to a backup.
+        uncompress: If True, the backup is assumed to be a gzipped tar and is
+            uncompressed before the database restoration.
+    """
+
+    if service_name == 'mongodb':
+        restore_mongodb(backup_path, uncompress)
+    elif service_name == 'mysql':
+        restore_mysql(backup_path, uncompress)
+
+
+def restore_mongodb(backup_path, uncompress=True):
+    """
+    Restore a MongoDB database from a backup.
+
+        backup_path: The absolute path to a backup.
+        uncompress: If True, the backup is assumed to be a gzipped tar and is
+            uncompressed before the database restoration.
+    """
+
+    logging.info('Restoring MongoDB from "{}"'.format(backup_path))
+
+    if uncompress:
+        backup_path = _uncompress(backup_path)
+
+    cmd = 'mongorestore {}'.format(backup_path)
+    ret = subprocess.call(cmd, shell=True)
+    if ret:  # if non-zero return
+        error_msg = 'Error occurred while restoring MongoDB backup'
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    logging.info('MongoDB successfully restored')
+
+
+def restore_mysql(backup_path, uncompress=True):
+    """
+    Restore a MySQL database from a backup.
+
+        backup_path: The absolute path to a backup.
+        uncompress: If True, the backup is assumed to be a gzipped tar and is
+            uncompressed before the database restoration.
+    """
+
+    logging.info('Restoring MySQL from "{}"'.format(backup_path))
+
+    if uncompress:
+        backup_path = _uncompress(backup_path)
+
+    cmd = 'mysqladmin -f drop edxapp'
+    ret = subprocess.call(cmd, shell=True)
+    if ret:  # if non-zero return
+        error_msg = 'Error occurred while deleting old mysql database'
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    cmd = 'mysqladmin -f create edxapp'
+    ret = subprocess.call(cmd, shell=True)
+    if ret:  # if non-zero return
+        error_msg = 'Error occurred while creating new mysql database'
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    cmd = 'mysql -D edxapp < {0}'.format(backup_path)
+    ret = subprocess.call(cmd, shell=True)
+    if ret:  # if non-zero return
+        error_msg = 'Error occurred while restoring mysql database'
+        logging.error(error_msg)
+        raise Exception(error_msg)
+
+    cmd = ('source /edx/app/edxapp/edxapp_env && /edx/bin/manage.edxapp '
+           'lms migrate --settings=aws_appsembler --delete-ghost-migrations')
```

Comment:
Ooh, thanks for pointing this out. What settings should we be using here @tkeemon ?